lxd/instances_put.go

```diff
@@ -102,15 +102,21 @@ func instancesPut(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(err)
 	}
 
-	var names []string
-	var instances []instance.Instance
 	for _, inst := range c {
 		if inst.Project().Name != projectName {
 			continue
 		}
 
-		// Only allow changing the state of instances the user has permission for.
+		// Check permission for all instances so that we apply the state change to all or none.
 		if !userHasPermission(entity.InstanceURL(inst.Project().Name, inst.Name())) {
+			return response.Forbidden(nil)
```

Comment:
Can we add a better error to the Forbidden response?

Reply:
I wasn't sure what the response should be as we're attempting not to make any resources discoverable...

Maybe `User does not have entitlement "can_update_state" for all instances`? But then they will know that there ane other instances in the project :thinking: